src/network/networkd-link.h

```diff
@@ -75,6 +75,7 @@ typedef struct Link {
         LinkOperationalState operstate;
         LinkCarrierState carrier_state;
         LinkAddressState address_state;
+        LinkOnlineState online_state;
```

Comment:
I think, it is better to initialize this with `_LINK_ONLINE_STATE_INVALID` in `link_new()`.